src/plugins/gopts/testapp.c

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file
+ *
+ * @brief Tests for specload plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <kdb.h>
+#include <kdbopts.h>
+#include <kdbplugin.h>
+
+#include <tests_plugin.h>
+
+#include "testdata.h"
+
+extern char ** environ;
+
+static KeySet * getSpec (const char * name, Key ** parentKey)
+{
+	if (strcmp (name, TEST_EMPTY) == 0)
+	{
+		*parentKey = keyNew ("spec/tests/gopts", KEY_END);
```

Comment:
Maybe it would make sense to create the parent key only once. As far as I can tell, each of the `if`-statements creates exactly the same key.